### PR TITLE
slack: unit tests for on-call messages

### PR DESCRIPTION
**Pull request by mastercactapus**
- [x] Identified the issue which this PR solves.
- [x] Read the [**CONTRIBUTING**](https://github.com/target/goalert/blob/master/CONTRIBUTING.md) document.
- [x] Code builds clean without any errors or warnings.
- [x] Added appropriate tests for any new functionality.
- [x] All new and existing tests passed.
- [x] Added comments in the code, where necessary.
- [x] Ran `make check` to catch common errors. Fixed any that came up.

**Description:**
Pulls the rendering logic out of the `Send` method for Slack on-call notifications into a unit-testable function and adds tests for edge cases.

### CODE REVIEW

**Review by arurao — DISMISSED** [review]

File: notification/slack/oncallnotification_test.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+)
+
+func TestRenderOnCallNotification(t *testing.T) {
+	slackUserIDs := map[string]string{"slack.1": "slack.1.ID", "slack.2": "slack.2.ID", "slack.3": "slack.3.ID"}
+	msg := notification.ScheduleOnCallUsers{
+		ScheduleName: "schedule.name",
+		ScheduleURL:  "schedule.url",
+	}
+
+	check := func(desc, expected string, users []string) {
```
Comment (Forfold) [inline notification/slack/oncallnotification_test.go:17]: Can you add a comment briefly going over the expected input/output for this test? I think it would be useful since `users` isn't used for the test assertion, but for setting the test/check up

e.g.
```suggestion
	// `expected` represents the string in plaintext as rendered by Slack
	// `users` represents the name for each expected user link.
	// If a user is prepended with "slack.", it should render with the following format: <@slack.xxx.ID>
	check := func(desc, expected string, users []string) {
```


I wasn't aware that it would translate from `slack.x` to `<@x.ID>`.,so oculd be valuable for others to know as well

Comment (mastercactapus) [reply notification/slack/oncallnotification_test.go]: There are no rules around prepending; it's just using the `map[string]string` above as an argument to the function that's being tested. I'll move it down and add a comment
```go
// renderOnCallNotificationMessage will render a message for Slack including links for the schedule and any users.
//
// If a user's ID is available in userSlackIDs, an `@` user mention will be used in place of a link to the GoAlert user's detail page.
func renderOnCallNotificationMessage(msg notification.ScheduleOnCallUsers, userSlackIDs map[string]string) string {
```

I made some of the naming more clear too.

Comment (Forfold) [reply notification/slack/oncallnotification_test.go]: Gotcha, thanks for clearing it up!

**Review by Forfold — APPROVED** [review]

**Review by arurao — APPROVED** [review]
